controllers/machinehealthcheck_controller_test.go

```diff
@@ -836,36 +820,55 @@ func TestMachineHealthCheck_Reconcile(t *testing.T) {
 			}
 
 			for i := range machines.Items {
-				if conditions.Get(&machines.Items[i], clusterv1.MachineOwnerRemediatedCondition) != nil {
-					unhealthyMachine = machines.Items[i].DeepCopy()
-					remediated++
+				if conditions.IsFalse(&machines.Items[i], clusterv1.MachineHealthCheckSuccededCondition) {
+					unhealthy++
 				}
 			}
 			return
-		}, timeout, 100*time.Millisecond).Should(Equal(1))
-
-		// Unpause the MachineSet reconciler.
-		machineSetPatch = client.MergeFrom(machineSet.DeepCopy())
-		delete(machineSet.Annotations, clusterv1.PausedAnnotation)
-		g.Expect(testEnv.Patch(ctx, machineSet, machineSetPatch)).To(Succeed())
+		}).Should(Equal(1))
 
-		// Make sure the Machine gets deleted.
+		// Make sure the Remeditaion Request is created.
 		g.Eventually(func() bool {
-			machine := unhealthyMachine.DeepCopy()
-			err := testEnv.Get(ctx, util.ObjectKey(unhealthyMachine), machine)
-			return apierrors.IsNotFound(err) || !machine.DeletionTimestamp.IsZero()
-		}, timeout, 100*time.Millisecond)
+			remediationTmpl := remediationTemplate.DeepCopy()
+			err := testEnv.Get(ctx, util.ObjectKey(infraRemediationTmpl), remediationTmpl)
+			return apierrors.IsNotFound(err)
+		}).Should(BeFalse())
```

Comment:
How we should verify here that created Remediation Request has an OwnerRef of unhealthy machine and RR name matches the unhealthy machine name.

Reply:
Same as below, we need to create an unstructured object that has the name and GVK of what we expect, then check that object instead of the template

Reply:
This might help. 
```golang
		ref := corev1.ObjectReference{
			APIVersion: "infrastructure.cluster.x-k8s.io/v1alpha3",
			Kind:       "InfrastructureRemediation",
		}

		obj := util.ObjectReferenceToUnstructured(ref)
		// Make sure the Remeditaion Request is created.
		g.Eventually(func() *unstructured.Unstructured {
			key := client.ObjectKey{
				Namespace: machines[0].Namespace,
				Name:      machines[0].Name,
			}
			err := testEnv.Get(ctx, key, obj)
			if err != nil {
				return nil
			} 
			return obj
		}, timeout, 100*time.Millisecond).ShouldNot(BeNil())
		g.Expect(obj.GetOwnerReferences()).To(HaveLen(1))
		g.Expect(obj.GetOwnerReferences()[0].Name).To(Equal(machines[0].Name))
                 // <-------------Test other things Annotations and Labels -------------->
```

Reply:
Thanks @wfernandes. We don't necessary need to verify anything else than OwnerRef and that it matches the unhealthy machine name . That is everything we are expecting here.